File: Haber/java haber/Haber.java
import java.util.ArrayList;
import java.util.List;

public class Haber {
    private Mail mail;
    private Bildirim bildirim;
    private List<Iletisim> iletisimListesi;

    public Haber(){
        iletisimListesi = new ArrayList<>();
        mail = new Mail();
        bildirim = new Bildirim();
        iletisimListesi.add(bildirim);
        iletisimListesi.add(mail);

    }

    public void bilgilendir(Kullanici kullanici, String mesaj){
        for (Iletisim iletisim : iletisimListesi) {
            iletisim.bilgiGonder(kullanici, mesaj);
        }
    }
    
}

File: Haber/csharp haber/Haber.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp_haber
{
    public class Haber
    {
        private Bildirim bildirim;
        private Mail mail;
        private List<Iletisim> iletisimListesi;

        public Haber(){
            iletisimListesi = new List<Iletisim>();
            mail = new Mail();
            bildirim = new Bildirim();
            iletisimListesi.Add(mail);
            iletisimListesi.Add(bildirim);
        }

        public void Bilgilendir(Kullanici kullanici, string mesaj){
            foreach (Iletisim iletisim in iletisimListesi){
                iletisim.BilgiGonder(kullanici, mesaj);
            }
        }
    }
}